packages/dds/tree/src/feature-libraries/sequence-field/compose.ts

```diff
@@ -179,8 +180,30 @@ function composeMarksIgnoreChild(
 	newMark: Mark,
 	moveEffects: MoveEffectTable,
 ): Mark {
+	if (isNoopMark(baseMark)) {
+		return newMark;
+	} else if (isNoopMark(newMark)) {
+		return baseMark;
+	}
+
+	if (isRename(baseMark) && isRename(newMark)) {
+		return { ...baseMark, idOverride: newMark.idOverride };
+	} else if (isRename(baseMark)) {
+		assert(isAttach(newMark) || isAttachAndDetachEffect(newMark), "Unexpected mark type");
+		return { ...newMark, cellId: baseMark.cellId };
```

Comment:
Are rename marks always supposed to have a cell ID? Do we assume that rename marks are never rebased?

Reply:
They are (same as Attach&Detach marks).